tests/integration/test_kinesis.py

```diff
@@ -236,6 +236,44 @@ def _get_shard_iterator(self, stream_name):
         return response.get("ShardIterator")
 
 
+@pytest.fixture
+def wait_for_stream_ready(kinesis_client):
+    def _wait_for_stream_ready(stream_name: str):
+        def is_stream_ready():
+            describe_stream_response = kinesis_client.describe_stream(StreamName=stream_name)
+            return describe_stream_response["StreamDescription"]["StreamStatus"] in [
+                "ACTIVE",
+                "UPDATING",
+            ]
+
+        poll_condition(is_stream_ready)
+
+    return _wait_for_stream_ready
+
+
+def test_get_records_next_shard_iterator(
+    kinesis_client, kinesis_create_stream, wait_for_stream_ready
+):
+    stream_name = kinesis_create_stream()
```

Comment:
nit suggestion: we could probably extend `kinesis_create_stream(..)` with a `wait_until_ready=False` parameter, to make the new fixture directly accessible/usable there. (not for this PR, only food for thought for future iterations)